api/authmethods/changestate.go

```diff
@@ -0,0 +1,65 @@
+package authmethods
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/boundary/api"
+)
+
+func (c *Client) ChangeState(ctx context.Context, authMethodId string, version uint32, state string, opt ...Option) (*AuthMethodUpdateResult, error) {
```

Comment:
any reason to allow an empty state parameter?

Reply:
No.  Added check and test for it.